WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -42,39 +55,68 @@ final class MainTabViewModel {
     ///
     func startObservingOrdersCount() {
         observeBadgeRefreshNotifications()
+        updateBadgeFromCache()
         requestBadgeCount()
     }
 }
 
 
 private extension MainTabViewModel {
+
+    /// Construct `ResultsController` with `siteID`
+    ///
+    func updateStatusResultsController(siteID: Int64) -> ResultsController<StorageOrderStatus> {
```

Comment:
Since this returns a new object, probably `createStatusResultsController` is a better fit. What do you think?